assets/pages/oneoff-contributions/components/oneoffContributionsPayment.jsx

```diff
@@ -10,46 +10,64 @@ import { Redirect } from 'react-router';
 import { routes } from 'helpers/routes';
 import StripePopUpButton from 'components/paymentButtons/stripePopUpButton/stripePopUpButton';
 import ErrorMessage from 'components/errorMessage/errorMessage';
-
-import { validateEmailAddress } from 'helpers/utilities';
-
 import type { ReferrerAcquisitionData } from 'helpers/tracking/acquisitions';
 import type { Participations } from 'helpers/abTests/abtest';
 import type { IsoCurrency } from 'helpers/internationalisation/currency';
 import type { Status } from 'helpers/switch';
-
-import { checkoutError, type Action } from '../oneoffContributionsActions';
+import { type UserFormFieldAttribute, formFieldIsValid } from 'helpers/checkoutForm/checkoutForm';
 import postCheckout from '../helpers/ajax';
-
+import { setFullNameShouldValidate, setEmailShouldValidate } from './contributionsCheckoutContainer/checkoutFormActions';
 
 // ----- Types ----- //
 
 type PropTypes = {|
   dispatch: Function,
   email: string,
+  setShouldValidateFunctions: Array<() => void>,
   error: ?string,
-  areAnyRequiredFieldsEmpty: boolean,
   amount: number,
   referrerAcquisitionData: ReferrerAcquisitionData,
-  checkoutError: (?string) => void,
   abParticipations: Participations,
   currencyId: IsoCurrency,
   isTestUser: boolean,
   isPostDeploymentTestUser: boolean,
   stripeSwitchStatus: Status,
   paymentComplete: boolean,
+  formFields: Array<UserFormFieldAttribute>
 |};
 
 
 // ----- Map State/Props ----- //
 
 function mapStateToProps(state) {
+
+  const fullName = {
+    value: state.page.user.fullName,
+    ...state.page.checkoutForm.fullName,
+  };
+
+  const email = {
+    value: state.page.user.email,
+    ...state.page.checkoutForm.email,
+  };
+
+  const fullNameFormField = {
+    value: fullName.value,
+    shouldValidate: state.page.checkoutForm.fullName.shouldValidate,
+    isValid: formFieldIsValid(fullName),
+  };
+  const emailFormField = {
+    value: email.value,
+    shouldValidate: state.page.checkoutForm.email.shouldValidate,
+    isValid: formFieldIsValid(email),
+  };
```

Comment:
If these are identical to what's in `mapStateToProps` in the `formFields` file, is it worth pulling them out into a shared helper?

Reply:
<3 this